### PR TITLE
Disable building Proton JavaScript bindings and other non-essential stuff

### LINKED ISSUE #2279: Proton dependency fails to build
```
mkdir -p /home/str4d/dev/zcash/zcash/depends/sources/download-stamps /home/str4d/dev/zcash/zcash/depends/sources
rm -f /home/str4d/dev/zcash/zcash/depends/sources/download-stamps/.stamp_fetched-proton-qpid-proton-0.17.0.tar.gz.hash
touch /home/str4d/dev/zcash/zcash/depends/sources/download-stamps/.stamp_fetched-proton-qpid-proton-0.17.0.tar.gz.hash
cd /home/str4d/dev/zcash/zcash/depends/sources/download-stamps; (test -f /home/str4d/dev/zcash/zcash/depends/sources/qpid-proton-0.17.0.tar.gz || ( mkdir -p /home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0 && echo Fetching proton... && ( wget --timeout=10 --tries=3 -nv -O "/home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0/qpid-proton-0.17.0.tar.gz.temp" "https://z.cash/depends-sources/qpid-proton-0.17.0.tar.gz" || wget --timeout=10 --tries=3 -nv -O "/home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0/qpid-proton-0.17.0.tar.gz.temp" "http://apache.cs.utah.edu/qpid/proton/0.17.0/qpid-proton-0.17.0.tar.gz" ) && echo "6ffd26d3d0e495bfdb5d9fefc5349954e6105ea18cc4bb191161d27742c5a01a  /home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0/qpid-proton-0.17.0.tar.gz.temp" > /home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0/.qpid-proton-0.17.0.tar.gz.hash && sha256sum -c /home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0/.qpid-proton-0.17.0.tar.gz.hash && mv /home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0/qpid-proton-0.17.0.tar.gz.temp /home/str4d/dev/zcash/zcash/depends/sources/qpid-proton-0.17.0.tar.gz && rm -rf /home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0 ))
Fetching proton...
2017-04-19 13:13:13 URL:https://z.cash/depends-sources/qpid-proton-0.17.0.tar.gz [946745/946745] -> "/home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0/qpid-proton-0.17.0.tar.gz.temp" [1]
/home/str4d/dev/zcash/zcash/depends/work/download/proton-0.17.0/qpid-proton-0.17.0.tar.gz.temp: OK
cd /home/str4d/dev/zcash/zcash/depends/sources; sha256sum qpid-proton-0.17.0.tar.gz >> /home/str4d/dev/zcash/zcash/depends/sources/download-stamps/.stamp_fetched-proton-qpid-proton-0.17.0.tar.gz.hash;
touch /home/str4d/dev/zcash/zcash/depends/sources/download-stamps/.stamp_fetched-proton-qpid-proton-0.17.0.tar.gz.hash
echo Extracting proton...
Extracting proton...
mkdir -p /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465
cd /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465; mkdir -p /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465 && echo "6ffd26d3d0e495bfdb5d9fefc5349954e6105ea18cc4bb191161d27742c5a01a  /home/str4d/dev/zcash/zcash/depends/sources/qpid-proton-0.17.0.tar.gz" > /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/.qpid-proton-0.17.0.tar.gz.hash &&  sha256sum -c /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/.qpid-proton-0.17.0.tar.gz.hash && tar --strip-components=1 -xf /home/str4d/dev/zcash/zcash/depends/sources/qpid-proton-0.17.0.tar.gz
/home/str4d/dev/zcash/zcash/depends/sources/qpid-proton-0.17.0.tar.gz: OK
touch /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/.stamp_extracted
echo Preprocessing proton...
Preprocessing proton...
mkdir -p /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465 /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/.patches-3fc2682d465
cd /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465;   sed -i '1i set(CMAKE_CXX_FLAGS " -std=c++11")' proton-c/bindings/cpp/CMakeLists.txt && sed -i '1i set(CMAKE_CXX_FLAGS " -std=c++11")' examples/cpp/CMakeLists.txt && sed -i '1i set(CMAKE_CXX_FLAGS " -std=c++11")' CMakeLists.txt && sed -i.old 's/qpid-proton SHARED/qpid-proton STATIC/' proton-c/CMakeLists.txt && sed -i.old 's/SASL/_DO_NOT_BUILD_SASL_/' proton-c/CMakeLists.txt && sed -i.old 's/qpid-proton-core SHARED/qpid-proton-core STATIC/' proton-c/CMakeLists.txt && sed -i.old 's/find_package(OpenSSL)/#find_package(OpenSSL)/' proton-c/CMakeLists.txt && sed -i.old 's/qpid-proton-cpp SHARED/qpid-proton-cpp STATIC/' proton-c/bindings/cpp/CMakeLists.txt && sed -i.old 's/DEFAULT_GO ON/DEFAULT_GO OFF/' proton-c/bindings/CMakeLists.txt && mkdir build
touch /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/.stamp_preprocessed
echo Configuring proton...
Configuring proton...
rm -rf /home/str4d/dev/zcash/zcash/depends/x86_64-unknown-linux-gnu; mkdir -p /home/str4d/dev/zcash/zcash/depends/x86_64-unknown-linux-gnu/lib; cd /home/str4d/dev/zcash/zcash/depends/x86_64-unknown-linux-gnu; 
mkdir -p /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/.
cd /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/.;     PKG_CONFIG_LIBDIR=/home/str4d/dev/zcash/zcash/depends/x86_64-unknown-linux-gnu/lib/pkgconfig PKG_CONFIG_PATH=/home/str4d/dev/zcash/zcash/depends/x86_64-unknown-linux-gnu/share/pkgconfig PATH=/home/str4d/dev/zcash/zcash/depends/x86_64-unknown-linux-gnu/native/bin:/home/str4d/app/tor-browser_en-US/bin:/home/str4d/app/signal/bin:/home/str4d/app/keepassx-2.0.2.tar.gz/bin:/home/str4d/app/keepassx-2.0.2/bin:/home/str4d/app/idea-IU-old/bin:/home/str4d/app/ideaIU-2016.2.tar.gz/bin:/home/str4d/app/idea-IU/bin:/home/str4d/app/i2p/bin:/home/str4d/app/heaptrack/bin:/home/str4d/app/games/bin:/home/str4d/app/Coati_Trial_0_8_0_Linux_64bit.tar.gz/bin:/home/str4d/app/Coati-Trial/bin:/home/str4d/app/BOB-one/bin:/home/str4d/app/android-studio/bin:/home/str4d/.cargo/bin:/home/str4d/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin   cd build; cmake .. -DCMAKE_CXX_STANDARD=11 -DCMAKE_INSTALL_PREFIX=/ -DSYSINSTALL_BINDINGS=ON -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DBUILD_PYTHON=OFF -DBUILD_PHP=OFF -DBUILD_JAVA=OFF -DBUILD_PERL=OFF -DBUILD_RUBY=OFF -DDEFAULT_GO=OFF
-- The C compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is GNU 5.4.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build type is "RelWithDebInfo" (has debug symbols)
-- PN_VERSION: 0.17.0 ()
-- Can't locate the valgrind command; no run-time error detection
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Could NOT find SWIG (missing:  SWIG_EXECUTABLE SWIG_DIR) 
-- Could NOT find Cyrus_DO_NOT_BUILD_SASL_ (missing:  CYRUS_SASL_LIBRARY CYRUS_SASL_INCLUDE_DIR) 
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for atoll
-- Looking for atoll - found
-- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE) 
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Found emscripten, using that to build JavaScript binding
-- Installing node.js package: ws@0.5.0
npm WARN enoent ENOENT: no such file or directory, open '/home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/package.json'
npm WARN 0.17.0-3fc2682d465 No description
npm WARN 0.17.0-3fc2682d465 No repository field.
npm WARN 0.17.0-3fc2682d465 No README data
npm WARN 0.17.0-3fc2682d465 No license field.
-- Installed node.js package: ws@0.5.0
-- Installing node.js package: jsdoc
npm WARN enoent ENOENT: no such file or directory, open '/home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/package.json'
npm WARN 0.17.0-3fc2682d465 No description
npm WARN 0.17.0-3fc2682d465 No repository field.
npm WARN 0.17.0-3fc2682d465 No README data
npm WARN 0.17.0-3fc2682d465 No license field.
-- Installed node.js package: jsdoc
-- JavaScript build type is "Release"
-- emscripten compilation environment:
-- CMAKE_C_COMPILER: /usr/bin/emcc
-- Looking for uuid_generate in uuid
-- Looking for uuid_generate in uuid - not found
-- Documentation Enabled. Using /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/node_modules/.bin/jsdoc to build JavaScript docs
-- Performing Test HAS_LONG_LONG
-- Performing Test HAS_LONG_LONG - Success
-- Performing Test HAS_STD_PTR
-- Performing Test HAS_STD_PTR - Success
-- Performing Test HAS_CPP11
-- Performing Test HAS_CPP11 - Success
-- Could NOT find Doxygen (missing:  DOXYGEN_EXECUTABLE) 
-- Cannot find ruby, skipping ruby tests
-- Found Libuv: /usr/lib/x86_64-linux-gnu/libuv.so  
-- Configuring done
CMake Error at proton-c/bindings/javascript/CMakeLists.txt:177 (add_library):
  Cannot find source file:

    /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/proton-c/src/object/object.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


CMake Error: CMake can not determine linker language for target: qpid-proton-bitcode
CMake Error: Cannot determine link language for target "qpid-proton-bitcode".
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_JAVA
    BUILD_PERL
    BUILD_PHP


-- Build files have been written to: /home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/build
funcs.mk:238: recipe for target '/home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/./.stamp_configured' failed
make: *** [/home/str4d/dev/zcash/zcash/depends/work/build/x86_64-unknown-linux-gnu/proton/0.17.0-3fc2682d465/./.stamp_configured] Error 1
make: Leaving directory '/home/str4d/dev/zcash/zcash/depends'
```

Looks like it is trying to build the JavaScript bindings, and failing for some reason.

```
$ gcc --version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.2 LTS"
$ emcc --version
emcc (Emscripten GCC-like replacement) 1.22.1 ()
```

Emscripten is installed for me because I attempted to build `libsodium.js` as part of #1113.

### PR DESCRIPTION
**Pull request by str4d**
Closes #2279.

### CODE REVIEW

**Comment by bitcartel** [comment]
@daira @str4d I've pushed a commit to this PR for review.  Cleans up building.  See commit message; no longer builds docs, examples, language bindings (except for cpp), also does not try and find packages which we know we don't need.

**Comment by bitcartel** [comment]
@zkbot try

**Comment by daira** [comment]
The ``sed`` commands look like they could be better expressed as a patch; that would be easier to review.

**Comment by bitcartel** [comment]
@str4d If I make a patch for this as @daira suggested, is it okay (since you opened this PR) if I squash all the commits and do a force push?

**Comment by bitcartel** [comment]
@zkbot try

**Comment by ageis** [comment]
@zkbot retry

File: depends/patches/proton/minimal-build.patch [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+ 
+-add_custom_target(docs)
+-add_custom_target(doc DEPENDS docs)
++#add_custom_target(docs)
++#add_custom_target(doc DEPENDS docs)
+ 
+ # Set the default SSL/TLS implementation
+-find_package(OpenSSL)
++#find_package(OpenSSL)
+ find_package(PythonInterp REQUIRED)
```
Comment (daira) [inline depends/patches/proton/minimal-build.patch]: Should this be disabled?

File: depends/patches/proton/minimal-build.patch [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+ 
+ foreach (LANG ${BINDING_LANGS})
+   set (SYSINSTALL_${LANG} OFF CACHE BOOL "Install ${LANG} bindings into interpreter specified location.")
+@@ -156,10 +157,10 @@ set (PROTON_SHARE ${SHARE_INSTALL_DIR}/proton-${PN_VERSION})
+ # End of variables used during install
+ 
+ # Check for valgrind here so tests under proton-c/ and examples/ can use it.
+-find_program(VALGRIND_EXE valgrind DOC "Location of the valgrind program")
++#find_program(VALGRIND_EXE valgrind DOC "Location of the valgrind program")
+ mark_as_advanced (VALGRIND_EXE)
```
Comment (daira) [inline depends/patches/proton/minimal-build.patch]: Comment this out as well.

File: depends/patches/proton/minimal-build.patch [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
++#add_custom_target(doc DEPENDS docs)
+ 
+ # Set the default SSL/TLS implementation
+-find_package(OpenSSL)
++#find_package(OpenSSL)
+ find_package(PythonInterp REQUIRED)
+-find_package(SWIG)
++#find_package(SWIG)
+ # FindSwig.cmake "forgets" make its outputs advanced like a good citizen
+ mark_as_advanced(SWIG_DIR SWIG_EXECUTABLE SWIG_VERSION)
```
Comment (daira) [inline depends/patches/proton/minimal-build.patch]: Comment this out as well.

File: depends/patches/proton/minimal-build.patch [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+ 
+ # See if Cyrus SASL is available
+-find_library(CYRUS_SASL_LIBRARY sasl2)
+-find_path(CYRUS_SASL_INCLUDE_DIR sasl/sasl.h PATH_SUFFIXES include)
+-find_package_handle_standard_args(CyrusSASL DEFAULT_MSG CYRUS_SASL_LIBRARY CYRUS_SASL_INCLUDE_DIR)
++#find_library(CYRUS_SASL_LIBRARY sasl2)
++#find_path(CYRUS_SASL_INCLUDE_DIR sasl/sasl.h PATH_SUFFIXES include)
++#find_package_handle_standard_args(CyrusSASL DEFAULT_MSG CYRUS_SASL_LIBRARY CYRUS_SASL_INCLUDE_DIR)
+ mark_as_advanced(CYRUS_SASL_LIBRARY CYRUS_SASL_INCLUDE_DIR)
```
Comment (daira) [inline depends/patches/proton/minimal-build.patch]: Comment this out as well.

File: depends/patches/proton/minimal-build.patch [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+ 
+ # Find saslpasswd2 executable to generate test config
+-find_program(SASLPASSWD_EXE saslpasswd2 DOC "Program used to make SASL user db for testing")
++#find_program(SASLPASSWD_EXE saslpasswd2 DOC "Program used to make SASL user db for testing")
+ mark_as_advanced(SASLPASSWD_EXE)
```
Comment (daira) [inline depends/patches/proton/minimal-build.patch]: Comment this out as well.

File: depends/patches/proton/minimal-build.patch [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+ if (EMSCRIPTEN_FOUND)
+   set (DEFAULT_JAVASCRIPT ON)
+ endif (EMSCRIPTEN_FOUND)
+@@ -37,7 +37,7 @@ if (CMAKE_CXX_COMPILER)
+ endif (CMAKE_CXX_COMPILER)
+ 
+ # Prerequisites for Go
+-find_program(GO_EXE go)
++#find_program(GO_EXE go)
+ mark_as_advanced(GO_EXE)
```
Comment (daira) [inline depends/patches/proton/minimal-build.patch]: Comment this out as well.

File: depends/patches/proton/minimal-build.patch [hunk 7]
```diff
@@ -0,0 +1,288 @@
+From 03f5fc0826115edbfca468261b70c0daf627f488 Mon Sep 17 00:00:00 2001
+From: Simon <simon@bitcartel.com>
+Date: Thu, 27 Apr 2017 17:15:59 -0700
+Subject: [PATCH] Only build minimal dependencies needed for cpp bindings.
```
Comment (daira) [inline depends/patches/proton/minimal-build.patch]: Add that this enables C++11 and builds a static library.

Comment (daira) [inline depends/patches/proton/minimal-build.patch]: ... or separate that into another patch maybe. (Does not block.)

File: depends/patches/proton/minimal-build.patch (hunk 2 above)
Comment (bitcartel) [inline depends/patches/proton/minimal-build.patch]: I think you mean that you want "#mark_as_advanced" ?  From cmake docs, 'mark_as_advanced' doesn't do anything with regards to the build, it will hide the variable from the CMake GUI.   I didn't comment it out as I wanted the patch to be as small as possible.

File: depends/patches/proton/minimal-build.patch (hunk 7 above)
Comment (bitcartel) [inline depends/patches/proton/minimal-build.patch]: I'll change the subject (and commit message) in a bit.

File: depends/patches/proton/minimal-build.patch (hunk 1 above)
Comment (bitcartel) [inline depends/patches/proton/minimal-build.patch]: No, if disabled, build fails.  There are generated source files which require Python.

File: depends/patches/proton/minimal-build.patch (hunk 7 above)
Comment (bitcartel) [inline depends/patches/proton/minimal-build.patch]: Fixed.

File: depends/patches/proton/minimal-build.patch (hunk 3 above)
Comment (bitcartel) [inline depends/patches/proton/minimal-build.patch]: See above about mark_as_advanced.

File: depends/patches/proton/minimal-build.patch (hunk 4 above)
Comment (bitcartel) [inline depends/patches/proton/minimal-build.patch]: Ditto

File: depends/patches/proton/minimal-build.patch (hunk 5 above)
Comment (bitcartel) [inline depends/patches/proton/minimal-build.patch]: Ditto

File: depends/patches/proton/minimal-build.patch (hunk 6 above)
Comment (bitcartel) [inline depends/patches/proton/minimal-build.patch]: Ditto

**Comment by bitcartel** [comment]
Checked CI server and buildbot tests passed.
http://ci.z.cash:8010/#/builders/14/builds/1
Note, this is a temporary link before CI servers get reset and rebuilt (at which point we can delete some of the zkbot related messages on this ticket)

**Comment by ageis** [comment]
retrying due to previous failure of Buildbot to tell Homu about the final build status and config update to fix the issue @zkbot retry

**Comment by ageis** [comment]
@zkbot try

**Comment by ageis** [comment]
@zkbot retry

**Comment by ageis** [comment]
@zkbot try

**Comment by str4d** [comment]
Testing new Homu and Buildbot setup.

@zkbot try

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by daira** [comment]
@zkbot retry

**Comment by daira** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot are you still there?

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
Testing stopselected:

@zkbot force

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
Testing force:

@zkbot try

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot clear

**Comment by str4d** [comment]
@zkbot clean

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot clean

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
@zkbot r+

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
@zkbot try

**Comment by str4d** [comment]
@zkbot retry

**Comment by str4d** [comment]
@zkbot r-

**Comment by str4d** [comment]
@zkbot force

**Comment by str4d** [comment]
Future readers: the various merges above were created while I was testing the CI. I have deleted the corresponding commands and responses to clean up the PR history.

**Comment by nathan-at-least** [comment]
I'm kicking this out of 1.0.9 in favor of #2362. Let's follow up on this after 1.0.9.

**Comment by bitcartel** [comment]
@nathan-at-least This should be included in 1.0.9 because it fixes potential build problems for people who *do* want to build proton. Otherwise some people will opt in to build and be confused as to why it doesn't.

**Comment by nathan-at-least** [comment]
If I understand correctly, this branch passes on our CI builder that *doesn't* have the other languages installed, so our CI system does not test the build fixes that this patch introduces. We need to set up a CI builder specifically to test those changes, and I don't believe we have time to do so today.

**Comment by nathan-at-least** [comment]
I filed #2365 to follow up on the CI need here.

**Comment by nathan-at-least** [comment]
I moved #2365 and this PR back into 1.0.9 low on the page, indicating lower priority than the other tickets in 1.0.9.

**Comment by ageis** [comment]
We should make a list of stuff to build without this patch:

openssl
go
sasl2
swig
ruby
doxygen

**Comment by bitcartel** [comment]
@ageis valgrind, emscripten, cyrus-sasl, perl, python

**Comment by nathan-at-least** [comment]
Just completed #2365, so:

@zkbot r+
